src/System.Private.ServiceModel/src/Internals/System/Runtime/TimeoutHelper.cs

```diff
@@ -31,38 +43,34 @@ public TimeoutHelper(TimeSpan timeout)
             _deadlineSet = (timeout == TimeSpan.MaxValue);
         }
 
-        // No locks as we expect this class to be used linearly. 
-        // If another CancellationTokenSource is created, we might have a CancellationToken outstanding
-        // that isn't cancelled if _cts.Cancel() is called. This happens only on the Abort paths, so it's not an issue. 
-        private void InitializeCancellationToken(TimeSpan timeout)
+        public async Task<CancellationToken> GetCancellationTokenAsync()
         {
-            if (timeout == TimeSpan.MaxValue || timeout == Timeout.InfiniteTimeSpan)
-            {
-                _cancellationToken = CancellationToken.None;
-            }
-            else if (timeout > TimeSpan.Zero)
+            if (!_cancellationTokenInitialized)
             {
-                _cts = new CancellationTokenSource();
-                _cancellationToken = _cts.Token;
-                TimeoutTokenSource.FromTimeout((int)timeout.TotalMilliseconds).Register(s_cancelOnTimeout, this);
-            }
-            else
-            {
-                _cancellationToken = new CancellationToken(true);
-            }
-            _cancellationTokenInitialized = true;
-        }
+                // No locks as we expect this class to be used linearly.
```

Comment:
Comment no longer applicable as the coalescing code will hand out the same token if called from multiple threads.